quickwit/quickwit-index-management/src/garbage_collection.rs

```diff
@@ -174,19 +192,28 @@ async fn delete_splits_marked_for_deletion(
             .with_update_timestamp_lte(updated_before_timestamp)
             .with_limit(DELETE_SPLITS_BATCH_SIZE);
 
-        let list_splits_result = protect_future(progress_opt, metastore.list_splits(query)).await;
+        let list_splits_request = match ListSplitsRequest::try_from_list_splits_query(query) {
```

Comment:
It's tempting to remove the `try_from`. If this fails, we have a bug somewhere.

Reply:
yeah... I would rather keep this for now.